api/src/wfl/executor.clj

```diff
@@ -428,6 +446,80 @@
      executor
      (jdbc/query tx [(format query details) status]))))
 
+(defn ^:private reference-id-to-workflow-records
+  "Find all submission IDs associated with `workflows`.
+  Return a mapping of the submissions' snapshot reference IDs
+  to their associated workflow records."
+  [tx {:keys [details] :as executor} workflows]
+  (throw-if-no-details-table tx executor)
+  (let [workflow-ids (util/to-quoted-comma-separated-list (map :uuid workflows))
+        _            (-> "%s Fetching submission IDs linked to workflows %s"
+                         (format (log-prefix executor) workflow-ids)
+                         log/debug)
+        ;; Presently Rawls doesn't support submitting a subset of a snapshot:
```

Comment:
Flagging this, if Rawls is updated to support submitting snapshots of the subset then this will need to change.